index.test.js

```diff
@@ -248,13 +269,31 @@ describe('express joi', function() {
         .expect(200)
     })
 
-    it('should pass an error to subsequent handler if it is asked', function() {
+    it('should pass an error to subsequent handler if it is asked', function(done) {
       const middleware = mod.response(schema, {
         passError: true
       })
-      return getRequester(middleware)
+      const res = getRequester(middleware, errorHandler)
         .get('/response/one')
-        .expect(500)
+        .expect(422)
+        .end(function(err, res) {
+          expect(res.body.type).to.contain('response')
+          expect(res.body.message).to.contain('ValidationError:')
+          done()
+        })
+    })
+
+    it('Allow error handler to work if error occured in request handler', function(done) {
+      const middleware = mod.response(schema, {
+        passError: true
+      })
+      const res = getRequester(middleware, errorHandler)
+        .get('/error')
+        .expect(422)
```

Comment:
This should be a 500 error, yet the test is passing. That's odd. Checking `res.statusCode` in the `end` block instead will do the trick.

I'll need to look into why supertest is not asserting it correctly since it seems to be the same for other tests...

Reply:
The errorHandler sets the status code to 422, so I think this works as expected